rest/.jvm/src/main/scala/io/udash/rest/RestServlet.scala

```diff
@@ -74,19 +74,21 @@ class RestServlet(
     // can't use request.getPathInfo because it decodes the URL before we can split it
     val pathPrefix = request.getContextPath.orEmpty + request.getServletPath.orEmpty
     val path = PlainValue.decodePath(request.getRequestURI.stripPrefix(pathPrefix))
+
     val query = request.getQueryString.opt.map(PlainValue.decodeQuery).getOrElse(Mapping.empty)
+
     val headersBuilder = IMapping.newBuilder[PlainValue]
     request.getHeaderNames.asScala.foreach { headerName =>
       if (!headerName.equalsIgnoreCase(CookieHeader)) { // cookies are separate, don't include them into header params
         headersBuilder += headerName -> PlainValue(request.getHeader(headerName))
       }
     }
     val headers = headersBuilder.result()
-    val cookiesBuilder = Mapping.newBuilder[PlainValue]
-    request.getCookies.opt.getOrElse(Array.empty).foreach { cookie =>
-      cookiesBuilder += cookie.getName -> PlainValue(cookie.getValue)
+
+    val cookies = request.getHeaders(CookieHeader).asScala.foldLeft(Mapping.empty[PlainValue]) {
```

Comment:
Does this mean that we cannot rely on `getCookies` anywhere, anymore? It might be hard to use with filters / interceptors.

Reply:
Any other solution?

Reply:
I'm happy with the way things work, in fact, your implementation is compliant with the Jetty guidelines for this change:

I just want the decoding to be done on the `Cookie` objects as opposed to parsing the whole header, so we can always rely on the way they're composed in other Jetty abstractions. This seems to be possible to do here (simplified code sample):
```scala
val cookies = request.getCookies.opt.getOrElse(Array.empty).foldLeft(Mapping.empty[PlainValue]) {
  (cookies, headerValue) => cookies.append(URLEncoder.decode(headerValue.getName, true), PlainValue(URLEncoder.decode(headerValue.getValue, true)))
}
```